File: src/com/juhnowski/onthology/World.java
package com.juhnowski.onthology;

import com.juhnowski.onthology.logical.LogicalPicture;

import java.util.ArrayList;

public abstract class World extends Reality{
    private Facts facts;
    Structure structure;
    private World(Structure structure){
        super(structure);
        this.structure = structure;
        isTheCase = true;
    }

    public World(Structure structure, Facts facts){
        super(structure);
        this.structure = structure;
        isTheCase = true;
        this.facts = facts;
    }

    public abstract LogicalPicture scetch();

    public Form<Substance<TheObject>> form;

    public MaterialProperties materialProperties;

    public Form unalterableForm;
}

File: src/com/juhnowski/onthology/World.kt
package com.juhnowski.onthology

import com.juhnowski.onthology.logical.LogicalPicture

import java.util.ArrayList

abstract class World : Reality {
    private val facts: Facts
    internal var structure: Structure<*>
    var picture: WorldPicture? = null
    var description: ArrayList<ElementaryProposition>? = null

    var form: Form<Substance<TheObject>>? = null

    var materialProperties: MaterialProperties? = null

    var unalterableForm: Form<*>? = null

    private constructor(structure: Structure<*>) : super(structure) {
        this.structure = structure
        isTheCase = true
    }

    constructor(structure: Structure<*>, facts: Facts) : super(structure) {
        this.structure = structure
        isTheCase = true
        this.facts = facts
    }

    abstract fun scetch(): LogicalPicture<*>

}
